src/utils.c

```diff
@@ -2011,3 +2068,127 @@ tls_config_clone(as_config_tls* clone, const as_config_tls* src)
 	clone->certfile = safe_strdup(src->certfile);
 }
 
+void
+sc_config_clone(sc_cfg* clone, const sc_cfg* src)
+{
+	memcpy(clone, src, sizeof(sc_cfg));
+	clone->addr = safe_strdup(src->addr);
+	clone->port = safe_strdup(src->port);
+	sc_tls_clone(&clone->tls, &src->tls);
+}
+
+void
+sc_config_destroy(sc_cfg* cfg)
+{
+	if (cfg->addr != NULL) {
+		cf_free((char*) cfg->addr);
+	}
+	
+	if (cfg->port != NULL) {
+		cf_free((char*) cfg->port);
+	}
+
+	sc_tls_destroy(&cfg->tls);
+
+	memset(cfg, 0, sizeof(sc_cfg));
+}
+
+void
+sc_tls_clone(sc_tls_cfg* clone, const sc_tls_cfg* src)
+{
+	memcpy(clone, src, sizeof(sc_tls_cfg));
+	clone->ca_string = safe_strdup(src->ca_string);
+}
+
+void
+sc_tls_destroy(sc_tls_cfg* cfg)
+{
+	if (cfg->ca_string != NULL) {
+		cf_free((char*) cfg->ca_string);
+	}
+
+	memset(cfg, 0, sizeof(sc_tls_cfg));
+}
+
+char* read_file_as_string(const char* path)
+{
+	FILE* fptr;
+	long flen;
+
+	fptr = fopen(path, "rb");
+	if (fptr == NULL) {
+		err("failed to open %s", path);
+		return NULL;
+	}
+
+	if (fseek(fptr, 0, SEEK_END) != 0) {
+		err("failed to seek to end of %s", path);
+		return NULL;
+	}
+
+	flen = ftell(fptr);
+	if (flen < 0) {
+		err("failed to get file length of %s", path);
+		return NULL;
+	}
+
+	rewind(fptr);
+
+	size_t fsize = (size_t) flen;
+	char* buf = (char*) cf_malloc(fsize + 1);
+	if (buf == NULL) {
+		err("failed to allocate memory for file buff, path: %s", path);
+		return NULL;
+	}
+
+	fread(buf, fsize, 1, fptr);
+	if (ferror(fptr)) {
+		cf_free(buf);
+		err("failed to read %s", path);
+		return NULL;
+	}
+
+	if (fclose(fptr) != 0) {
+		cf_free(buf);
+		err("failed closing %s", path);
+		return NULL;
+	}
+
+	buf[fsize] = 0;
+
+	return buf;
+}
+
+int
+get_and_set_secret_arg(sc_client* sc, char* path, char** res, bool* is_secret) {
```

Comment:
Why is it called `get_and_set`? It looks like it only gets?

Reply:
renamed